kcidb/db/__init__.py

```diff
@@ -259,6 +260,30 @@ def complement(self, data):
         return self.query(ids=ids, children=True, parents=True)
 
 
+class DBHelpAction(argparse.Action):
+    """Argparse action outputting database string help and exiting."""
+    def __init__(self,
+                 option_strings,
+                 dest=argparse.SUPPRESS,
+                 default=argparse.SUPPRESS,
+                 help=None):
+        super().__init__(
+            option_strings=option_strings,
+            dest=dest,
+            default=default,
+            nargs=0,
+            help=help)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        print("Specify database as <DRIVER>:<PARAMS>. " +
+              "Driver options and associated params (if any) " +
+              "are listed below.\n")
+        for title, driver in DRIVER_TYPES.items():
+            print(title.upper() + " DRIVER \nPARAMS:\n" +
+                  driver.PARAMS_DOC)
```

Comment:
Calling the iterator variable `title` is a little imprecise. This is a driver's name, and we're *using* it to make a section title, so it's better to call it just `name`.

Then, at the moment, driver names accepted by the `-d`/`--database` option are case-sensitive (see `kcidb.db.Client.__init__()`). So we should be careful with making the only mention of a driver's name uppercase in our help message, because people could assume it should be supplied as such. We should either keep it the original case, or explicitly mention how it should be spelled. I'd say for now we can just print the driver headers something like this:
```
"json" driver
-------------
```
Further on, I think it would be good to give a brief description of each driver and its purpose, not just its parameters, so I think we need to switch from having parameter-specific docs in the drivers in `PARAMS_DOC`, to just general-purpose description including the parameter description in e.g. `DOC`.

In the end this loop could be changes something like this:
```suggestion
        for name, driver in DRIVER_TYPES.items():
            print(f"\n{name!r} driver\n" + 
                  "-" * (len(name) + 7) + "\n" +
                  driver.DOC)
```